### PR TITLE
adding crabserver grok to logstash

### PR DESCRIPTION
**Pull request by novicecpp**
https://its.cern.ch/jira/projects/CMSMONIT/issues/CMSMONIT-458

Add crabserver grok to cmsweb's logstash.
From what I understand, we (CRAB team) will use logstash/es-monit instances to build dashboard like [CRAB timber](https://monit-grafana.cern.ch/d/qUVV6S0Gk/crab-timber-pods?orgId=11&refresh=1h), but data will come from `crabserver` cherrypy logs instead.

I will open another PR for new crabserver sidecar.

### CODE REVIEW

**Comment by novicecpp** [comment]
@muhammadimranfarooqi @mrceyhun Please review.

**Comment by novicecpp** [comment]
From log line Stefano provided:
```
[06/Mar/2022:22:50:49] crabserver-766d664fbf-kg4wn 188.184.75.95 "POST /crabserver/prod/filetransfers HTTP/1.1" 200 OK [data: 1116 in 20 out 4865 us ] [auth: OK "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=sobarman/CN=839376/CN=Soumyadip Barman" "" ] [ref: "" "CRABSchedd/py3.220225" ]
```

This is output of format `message`
```json
{
  "agent_id": "a7e865a7-df5d-470d-a153-cff6a4e21775",
  "api": "filetransfers",
  "backend": "crabserver-766d664fbf-kg4wn",
  "bytes_received": "20",
  "bytes_sent": "1116",
  "client": "CRABSchedd/py3.220225",
  "clientip": "188.184.75.95",
  "cmsweb_cluster": "NA",
  "cmsweb_env": "NA",
  "cmsweb_log": "/logs/stefano.log",
  "code": 200,
  "dn": "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=sobarman/CN=Soumyadip Barman",
  "ephemeral_id": "23c2301d-d85d-4ecb-98dd-1cc5dbf44a1b",
  "hostname": "6c7e5eb25105",
  "httpversion": "HTTP/1.1",
  "message": "[06/Mar/2022:22:50:49] crabserver-766d664fbf-kg4wn 188.184.75.95 POST /crabserver/prod/filetransfers HTTP/1.1 200 OK [data: 1116 in 20 out 4865 us ] [auth: OK /DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=sobarman/CN=839376/CN=Soumyadip Barman  ] [ref:  CRABSchedd/py3.220225 ]",
  "method": "POST",
  "producer": "cmswebk8s",
  "rec_date": "2022-03-06T22:50:49.000Z",
  "rec_timestamp": "1646607049",
  "request": "/crabserver/prod/filetransfers",
  "time_spent_ms": "4865",
  "type": "crabserver",
  "uri_params": "",
  "uri_path": "/prod/filetransfers"
}
```

**Comment by mrceyhun** [comment]
@novicecpp could you please share more log lines to test grok. You can share via MM or in Jira ticket if logs are private.

File: kubernetes/cmsweb/monitoring/logstash.conf [hunk 1]
```diff
@@ -78,6 +78,26 @@ filter {
...
+      grok { match => { "uri_path" => '/.*/%{DATA:api}$' } }
+      date {
+         match => [ "tstamp", "dd/MMM/yyyy:HH:mm:ss" ]
+         target => "date_object"
+      }
+      ruby {
+         code => "event.set('rec_timestamp',event.get('date_object').to_i)
+                  event.set('rec_date',event.get('date_object'))
+                 "
+      }
```
Comment (mrceyhun) [inline kubernetes/cmsweb/monitoring/logstash.conf:99]: @novicecpp 
There is a  "timestamp" keyword of MONIT (like producer, type). They set `metadata.timestamp` as `timestamp` value if we set it. `metadata.timestamp` is the default date time of ElasticSearch index in Kibana. If you did not set it as you did, MONIT will set their own timestamp which will be the time when our data is reached to MONIT.

So, you can set timestamp with the event time like we did in "frontend" tag. You can use:
```
ruby {
         code => "event.set('timestamp',(event.get('date_object').to_f * 1000).to_i)
                  event.set('rec_date',event.get('date_object'))
                 "
      }

# And, add timestamp to message part in the output:
 message => '{"producer": "cmswebk8s", "type": "crabserver", ... , "timestamp": "%{timestamp}"}'

# Remove "rec_timestamp":"%{rec_timestamp}",
# rec_timestamp is not necessary in your case
```

Date fields will be like in ES:
`data.@timestamp`: Logstash timestamp
`metadata.kafka_timestamp`: MONIT kafka timestamp
`metadata.timestamp`: Event/Log time which we parsed from log

[the remark this answers is not in the document]
Comment (mrceyhun) [reply kubernetes/cmsweb/monitoring/logstash.conf:99]: Btw, the decision is up to you but it's convenient to use metadata.timestamp as log time in Kibana.

Other than this, I tested your settings and it's ready to deploy.

[the remark this answers is not in the document]
Comment (mrceyhun) [reply kubernetes/cmsweb/monitoring/logstash.conf:99]: Btw, the decision is up to you but it's convenient to use metadata.timestamp as log time in Kibana.

Other than this, I tested your settings and it's ready to deploy.

**Pull request merged**